### ISSUE TITLE
How to override default keybinding

### ISSUE THREAD

**Issue by schtucks** [KillerCodeMonkey/ngx-quill#1702]
I have problems with overriding the default keybinding enter. In my case, when pressing the Enter key, a function should be executed and not the default behavior linebreak. I have already spent hours reading the documentation of Quill itself and browsing all possible related issues in Quill's git and also in this git but found nothing that would have worked for me. I found kind of a workaround where I just added a listener to the (keydown.enter) event on my quill editor tag and then just submit the message and then empty the quill editor. However, this is more of a temporary solution. I also looked through the examples you provided. If I bind a key that doesn't have a default binding it works fine, but unfortunately not if one already exists. Now I have the big question how to unbind default keys in ngx-quill properly. 

Could you provide me a code sample?

Best regards

**Comment by KillerCodeMonkey** [comment]
Just do it the native quill way. Pass the bindings to the local or global
ngx-quill config, or use addKeyBinding of the quilljs instance you get in
each output or from the editor component itself.

Overwriting existing key bindings does not work when using the
addKeyBinding API.

You should check the quilljs sourcecode which keycode and config is used
for enter.
https://github.com/quilljs/quill/blob/1.3.7/modules/keyboard.js#L40

stax ***@***.***> schrieb am Mi., 21. Juni 2023, 12:47:

**Comment by schtucks** [comment]
We tested: 
`    QuillModule.forRoot({
			modules: {
        keyboard: {
          bindings: [
            {
              key: '13',
              handler: function(){
                console.log("XXX");
              }
            }
          ]
        }
			}
		})`

It don't change the default behaviour of quill?

**Comment by KillerCodeMonkey** [comment]
Key: 13 (as number) sorry

Bengt Weiße ***@***.***> schrieb am Mi., 21. Juni 2023, 16:38:

**Comment by schtucks** [comment]
Yes, that worked! Now when you hit enter the default behavior was removed. 

But unfortunately the rebinding of the key doesn't work ...

In my Quill Component: 
```
modules = {};

  constructor() {
    this.modules = {
      formula: true,
      keyboard: {
        addBinding: {
          key: 13,
          handler: () => { this.onSendMessageClick(); }
        }
      }
    }
  }
```
Isn't that the proper way?

**Comment by KillerCodeMonkey** [comment]
You can not simply use flat arrow function.

Quilljs does not know anything of angular and the context it is used.

Try

Constructor() {
  this.onSendMessageClick = this.onSendMessageClick.bind(this)
}

With that you are setting the call bindings to the current this

stax ***@***.***> schrieb am Do., 22. Juni 2023, 07:51:

**Comment by schtucks** [comment]
I referenced to one of your answeres on another issue: [here](https://github.com/KillerCodeMonkey/ngx-quill/issues/112)

Heres the code of your example:
```
@Component({...})
class xxx {
  modules = {};

  constructor() {
    this.modules = {
      formula: true,
      keyboard: {
        addBinding:{
          key: 'B',
          handler: () => { this.bHandler(); }
        }
     };
  }

  bHandler () {
    console.log('hi');
  }
}
```
You do it there the same way ... 

I tried: 
```
constructor() {
    this.modules = {
      keyboard: {
        addBinding: {
          key: '13',
          handler: () => {}
        }
      }
    }
    this.onSendMessageClick = this.onSendMessageClick.bind(this)
  }
```
And i also tried it on the created event: 

```
created(event: Quill) {
    this.quill = event;
    this.quill.keyboard.addBinding(
      { key: '13'},
      (x,y) => {
        console.log("XXXX"); 
      }
    );
  }
```

**Comment by KillerCodeMonkey** [comment]
```TS
@Component({...})
class xxx {
  modules = {};

  constructor() {
    // Bind context of your handler to your class instance - the binding callback is called inside the quilljs plain js context not in your angular component context
    this.bHandler = his.bHandler.bind(this);

    this.modules = {
      formula: true,
      keyboard: {
        addBinding:{
          key: 'B',
          handler: () => { this.bHandler(); }
        }
     };
  }

  bHandler () {
    console.log('hi');
  }
}

```

**Comment by KillerCodeMonkey** [comment]
did you try it?
